praw/models/redditors.py

```diff
@@ -67,6 +75,36 @@ def stream(
 
         Keyword arguments are passed to :func:`.stream_generator`.
 
-        :returns Redditor profiles, which are a type of :class:`.Subreddit`.
+        :returns: Redditor profiles, which are a type of :class:`.Subreddit`.
         """
         return stream_generator(self.new, **stream_options)
+
+    def user_summaries(
+        self, ids: Iterable[str]
+    ) -> Generator[UserSummary, None, None]:
+        """Get user summary data by redditor IDs.
+
+        :param ids: An iterable of redditor fullname IDs.
+        :returns: A iterator producing types.SimpleNamespace objects.
+
+        Each ID must be prefixed with "`t2_`".
+
+        Invalid IDs are ignored by the server.
+        """
+        iterable = iter(ids)
+        while True:
+            chunk = list(islice(iterable, 100))
+            if not chunk:
+                break
+
+            params = {"ids": ",".join(chunk)}
+            try:
+                results = self._reddit.get(
+                    API_PATH["user_by_fullname"], params=params
+                )
+            except prawcore.exceptions.NotFound:
+                # None of the given ids matched any Redditor.
+                continue
+
+            for fullname, user_data in results.items():
+                yield UserSummary(fullname=fullname, **user_data)
```

Comment:
Why can we not yield redditor instances? If we give them id and username, you should be able to resolve those without fetching. This will require an update to RedditBase.

Reply:
As I’ve expressed before, I’m straight up not super keen about the whole lazy-load feature of most PRAW objects. If users see this method returns `Redditor` objects they will try to use attributes they think they know these objects have and either it will lead to unexpected `AttributeError`s, or worse, will too easily unintentionally cause them to silently make loads of requests. Either way, It’s a big lie.

If I make this return `Redditor` objects then I’m going to have to document which attributes are ‘safe’ to use, and that’s really ugly.

Overmore, this is a bulk processing function. Space is a consideration. `SimpleNamespace` is pretty much as lightweight of a container as you can get.

Reply:
But the whole point of the function is to return usernames from fullnames. Would that not be a reasonable assumption that you can see the fullname and the username?

Reply:
I still think it should be returned, because what if I want to do a karma check on a list of fullnames? Then I'd have to create Redditor instances anyways. It should either return redditor objects or a dict -- or maybe have a switch for what to return.

Reply:
I’ve stated the caveats. You’re trading a tiny bit less compact code for setting up users with potential confusion, with worse space performance. It’s not good design.

> or maybe have a switch for what to return.

Please don’t even.